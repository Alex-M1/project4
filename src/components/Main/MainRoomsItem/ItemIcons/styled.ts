import styled from 'styled-components';

export const StIcon = styled.span`
    width: 40px;
    padding-top: 5px;
    justify-self: center;
`;
