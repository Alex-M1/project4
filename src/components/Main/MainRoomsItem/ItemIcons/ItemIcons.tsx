import React from 'react';
import { StIcon } from './styled';

interface IProps {
  src: string,
  alt: string,
}

const ItemIcons: React.FC<IProps> = ({ src, alt }) => {
  return (
    <StIcon>
      <img src={src} alt={alt} />
    </StIcon>
  );
};

export default ItemIcons;
