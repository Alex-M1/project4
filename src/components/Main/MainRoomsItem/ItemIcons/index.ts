import ItemIcons from './ItemIcons';

export default ItemIcons;
