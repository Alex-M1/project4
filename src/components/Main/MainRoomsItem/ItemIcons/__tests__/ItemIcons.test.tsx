import { mount, shallow } from 'enzyme';
import React from 'react';

import ItemIcons from '../ItemIcons';

describe('ItemIcons', () => {
  let props;
  beforeEach(() => {
    props = {
      src: 'string',
      alt: 'string',
    };
  });
  it('Should match snapshot', () => {
    const component = shallow(<ItemIcons {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StIcon', () => {
    const component = mount(<ItemIcons {...props} />);
    expect(component.find('styled__StIcon')).toHaveLength(1);
  });
  it('should render img', () => {
    const component = mount(<ItemIcons {...props} />);
    expect(component.find('img')).toHaveLength(1);
  });
});
