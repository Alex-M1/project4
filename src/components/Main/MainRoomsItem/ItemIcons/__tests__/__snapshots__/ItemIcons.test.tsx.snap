// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemIcons Should match snapshot 1`] = `"<span class=\\"styled__StIcon-chPeWD cDIvgb\\"><img src=\\"string\\" alt=\\"string\\"/></span>"`;
